File: Models/Document.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations.Schema;

namespace think_agro_metrics.Models
{
    [Table("Documents")]
    public class Document
    {

        public long DocumentID { get; set; }
        public string Name { get; set; }
        public string DocumentName { get; set; }
        public string Extension { get; set; }
        public string Link { get; set; }
        public DateTime Date { get; set; }
        public string Code { get; set; }
        public long RegistryID { get; set; }
        public Document() {
            Date = DateTime.Today;
        }


    }
}

File: ClientApp/src/app/shared/models/document.ts
export class Document {
  documentID: number;
  name: string;
  documentName: string;
  extension: string;
  link: string;
  date: Date;
  code: string;

  constructor() { }

  fromJSON(json) {
    for (const propName of json) {
      this[propName] = json[propName];
    }

    return this;
  }
}
